### PR TITLE
Add rename function for images after doing in place upgrade

### PR DESCRIPTION
**Pull request by MiriamDeng**
ID:2141163
Signed-off-by: MiriamDeng <mdeng@redhat.com>

### CODE REVIEW

**Comment by MiriamDeng** [comment]
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.q35: STARTED
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.q35: PASS (5767.49 s)

**Comment by MiriamDeng** [comment]
[stdlog] 2023-01-06 07:01:07,632 avocado.utils.process INFO | Running 'cp --reflink=auto /root/avocado/data/avocado-vt/images/rhel79-64-virtio-scsi.qcow2 /root/avocado/data/avocado-vt/images/rhel880-64-virtio-scsi.qcow2'
[stdlog] 2023-01-06 07:01:07,862 avocado.utils.process INFO | Command 'cp --reflink=auto /root/avocado/data/avocado-vt/images/rhel79-64-virtio-scsi.qcow2 /root/avocado/data/avocado-vt/images/rhel880-64-virtio-scsi.qcow2' finished with 0 after 0.225923587s
[stdlog] 2023-01-06 07:01:07,862 avocado.utils.process INFO | Running 'cp --reflink=auto /root/avocado/data/avocado-vt/avocado-vt-vm1_rhel79-64-virtio-scsi_qcow2_filesystem_VARS.fd /root/avocado/data/avocado-vt/avocado-vt-vm1_rhel880-64-virtio-scsi_qcow2_filesystem_VARS.fd'
[stdlog] 2023-01-06 07:01:07,867 avocado.utils.process INFO | Command 'cp --reflink=auto /root/avocado/data/avocado-vt/avocado-vt-vm1_rhel79-64-virtio-scsi_qcow2_filesystem_VARS.fd /root/avocado/data/avocado-vt/avocado-vt-vm1_rhel880-64-virtio-scsi_qcow2_filesystem_VARS.fd' finished with 0 after 0.000618387s

**Comment by MiriamDeng** [comment]
@PaulYuuu @fbq815 could you please help to review the patch ? Thanks.

**Comment by fbq815** [comment]
JOB ID     : 36a9fefd7915c8df408073f5d0d4e9d87430acf2
JOB LOG    : /root/avocado/job-results/job-2023-01-14T05.23-36a9fef/job.log
 (1/2) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/2) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (508.08 s)
 (2/2) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.s390x.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.s390-virtio: STARTED
 (2/2) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.s390x.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.s390-virtio: PASS (593.42 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-01-14T05.23-36a9fef/results.html
JOB TIME   : 1103.08 s

LGTM

File: qemu/tests/in_place_upgrade.py [hunk 1]
```diff
@@ -90,8 +94,25 @@ def run(test, params, env):
...
     finally:
-        if upgrade_test.session:
-            upgrade_test.session.close()
-        vm.destroy()
+        vm.graceful_shutdown(timeout=300)
+        try:
+            rt_dir = data_dir.get_data_dir() + "/"
+            image_format = params.get("image_format")
+            old_name = params.get("image_name")
+            old_name = "%s.%s" % (old_name, image_format)
```
Comment (PaulYuuu) [inline qemu/tests/in_place_upgrade.py:105]:
```suggestion
            image_name = params.objects("images")[0]
            image_params = params.object_params(image_name)
            old_name = storage.get_image_filename(
                data_image_params, data_dir.get_data_dir())
```

File: qemu/tests/in_place_upgrade.py [hunk 2]
```diff
@@ -90,8 +94,25 @@ def run(test, params, env):
...
+            process.run(params.get("image_clone_command") %
+                        (rt_dir + old_name, rt_dir + upgraded_name))
+            if "ovmf" in params.get("id"):
+                ovmf_file_old = rt_dir + params.get("main_vm") + "_"
+                ovmf_file_old += params.get("image_name")[7:] + "_"
+                ovmf_file_old += params.get("image_format")
+                ovmf_file_old += "_filesystem_VARS.fd"
+                ovmf_file = ovmf_file_old.replace(pre_rhel_ver, post_rhel_ver)
+                process.run(params.get("image_clone_command") %
+                            (ovmf_file_old, ovmf_file))
```
Comment (PaulYuuu) [inline qemu/tests/in_place_upgrade.py:116]: Not sure we need to handle this. firmware file will automatically be generated if it doesn't exist, and the format depends on the filesystem.

File: qemu/tests/in_place_upgrade.py (hunk 1 above)
Comment (MiriamDeng) [reply qemu/tests/in_place_upgrade.py:105]: Thanks for reviewing,.
Honestly,your way  need to import libs such as ,from virttest import qemu_storage,
from virttest import utils_disk and so on so forth, honestly, it's just a simple image backup&rename, your steps seem more complicated. Correct me if I was wrong, thank you !

File: qemu/tests/in_place_upgrade.py (hunk 2 above)
Comment (MiriamDeng) [reply qemu/tests/in_place_upgrade.py:116]: Yes, will remove it. Thanks.

File: qemu/tests/in_place_upgrade.py (hunk 1 above)
Comment (MiriamDeng) [reply qemu/tests/in_place_upgrade.py]: Updated, please help to review it  thanks. @PaulYuuu @fbq815

File: qemu/tests/in_place_upgrade.py [hunk 3]
```diff
@@ -90,8 +95,17 @@ def run(test, params, env):
...
     finally:
-        if upgrade_test.session:
-            upgrade_test.session.close()
-        vm.destroy()
+        vm.graceful_shutdown(timeout=300)
+        try:
+            image_name = params.objects("images")[0]
+            image_params = params.object_params(image_name)
+            old_name = storage.get_image_filename(image_params,
+                                                  data_dir.get_data_dir())
```
Comment (PaulYuuu) [inline qemu/tests/in_place_upgrade.py:106]: `data_dir.get_data_dir()` --> `params.get("images_base_dir", data_dir.get_data_dir())`

Comment (MiriamDeng) [reply qemu/tests/in_place_upgrade.py]: Thanks, updated !

**Comment by MiriamDeng** [comment]
(2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.8.0.x86_64.io-github-autotest-qemu.in_place_upgrade.without_rhsm_rhel8_to_rhel9.q35: PASS (5106.34 s)

File: qemu/tests/in_place_upgrade.py (hunk 3 above)
Comment (MiriamDeng) [reply qemu/tests/in_place_upgrade.py]: Could you help to review it again ? Thanks !

**Review by PaulYuuu — APPROVED** [review]

**Review by fbq815 — APPROVED** [review]

**Pull request merged**